### PR TITLE
Enhancement: Move test data for verification into files

### PR DESCRIPTION
**Pull request by Tigh-Gherr**

### CODE REVIEW

**Comment by Tigh-Gherr** [comment]
It would be nice to move the merkle proofs and txs used in envelope creation out to files, but the same tx in one test is sometimes set to be nil in another to test edge cases, and the same with merkle proofs.

This means the removal is going to be a bit more involved, so it can wait.

File: spv/createenvelope_test.go [hunk 1]
```diff
@@ -1133,7 +808,13 @@ func TestSPVEnvelope_CreateEnvelope(t *testing.T) {
 			if test.expErr == nil {
 				assert.NoError(t, err)
 				assert.NotNil(t, envelope)
-				assert.Equal(t, test.exp, *envelope)
+
+				f, err := os.Open("../testing/data/spv/create/" + test.expFile + ".json")
```
Comment (theflyingcodr) [inline spv/createenvelope_test.go:812]: Use fs.Embed instead - saves having to read files etc and can also be used to auto gen these too I think

**Review by theflyingcodr — CHANGES_REQUESTED** [review]
Consider using fs.Embed

**Review by jadwahab — APPROVED** [review]

File: spv/createenvelope_test.go (hunk 1 above)
Comment (Tigh-Gherr) [reply spv/createenvelope_test.go]: Good shout

Comment (Tigh-Gherr) [reply spv/createenvelope_test.go]: Undid this as it breaks compatibility with versions <1.16, which isn't worth in for cleaner test code IMO

Comment (theflyingcodr) [reply spv/createenvelope_test.go]: I think we should use the same compatibility as the go team for libs and only support that last 2 versions of go, in this case 1.17 and 1.16 [pub](https://golang.org/doc/devel/release#policy)

This is why I tend to only have test for last 2 versions

Comment (Tigh-Gherr) [reply spv/createenvelope_test.go]: Our build system supports the passed 3 tho

Comment (Tigh-Gherr) [reply spv/createenvelope_test.go]: Chatted offline, 1.15 is going the way of the dodo

**Review by theflyingcodr — APPROVED** [review]
LGTM
